### ISSUE TITLE
Can't run s3 writing example on 0.3

### ISSUE THREAD

**Issue by arokem** [richford/cloudknot#148]
I see this in the CloudFormation console: 

<img width="1637" alt="screen shot 2018-07-11 at 3 14 34 pm" src="https://user-images.githubusercontent.com/118582/42602407-314c8600-851e-11e8-8171-7ebc25f7b3ae.png">

**Comment by richford** [comment]
So under the previous versions of cloudknot, the user could supply a policy name. The new version requires them to supply a policy ARN. We can do two things to fix this issue:

1. Continue to require the user to supply a policy ARN. In this case, we update the S3 example to supply an ARN.
2. Modify `Pars` to do some validation of the `policies` input. If the policies are already ARNs, then just pass them to CloudFormation. If they are names, then we have to use boto3 to retrieve the corresponding ARNs.

The advantage of approach 2 is that the user can mix and match policy ARNs and policy names. The policy names are shorter to write. The advantage of approach 1 is that it will be faster not to have to look up the ARNs. If all we have is the policy name, then I think we'd have to use boto3's [`list_policies`](https://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.list_policies) to list all of the policies and then filter the results to find the one that matches the name. This would add some time to `Knot.__init__`.

I'm leaning toward approach 1. Although it'd be nice for the user to be able to use policy names, I'm hesitant to add time to the already lengthy `Knot.__init__` and my suspicion is that if the user is looking up a policy name, they can get the policy ARN will minimal additional effort.

@arokem, your thoughts?

**Comment by arokem** [comment]
I agree that if approach 2 doesn't add too much to startup time, it is preferable. 

If it's just a matter of running `list_policies`, then that seems fairly fast: 

```
In [1]: import boto3

In [2]: client = boto3.client('iam')

In [3]: %timeit client.list_policies()
185 ms ± 34.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

**Comment by richford** [comment]
Solution is sitting in #149. FWIW, the results of `list_policies()` are paginated so I think a better benchmark would be:

```
In [1]: import boto3

In [2]: iam = boto3.client('iam')

In [3]: %%timeit
        # Get all AWS policies
        response = iam.list_policies()
        aws_policies = {d['PolicyName']: d['Arn'] for d in response.get('Policies')}

        # If results are paginated, continue appending to aws_policies,
        # using `Marker` to tell next call where to start
        while response['IsTruncated']:
            response = iam.list_policies(Marker=response['Marker'])
            aws_policies.update(
                {d['PolicyName']: d['Arn'] for d in response.get('Policies')}
            )

727 ms ± 44.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Still fairly fast. I tested #149. I'll remove the "WIP" title. And if/once you get a chance to test drive it, I'll go ahead and merge.

**Comment by arokem** [comment]
Works great. Merge whenever you are ready.
